### PR TITLE
FEAT: public pages sitemap

### LINKED ISSUE #101: Dynamic sitemap for public page
**Is your feature request related to a problem? Please describe.**
We already have a static sitemap for a static URL, but still doesn't have for the dynamic one.

We need generate the dynamic sitemap for public question/question submission page (check on https://www.tanyaaja.in/eksplor), e.g:

- https://www.tanyaaja.in/p/mzp
- https://www.tanyaaja.in/p/zain
- https://www.tanyaaja.in/p/fatihkalifa

**Describe the solution you'd like**

- Select all users, especially the one that already marked as public users
- Generate the sitemap
- Adding the path to the sitemap index
- https://nextjs.org/learn/seo/crawling-and-indexing/xml-sitemaps

### PR DESCRIPTION
**Pull request by adjieguntoro-dev**
Closes #101

## Description

enable dynamic sitemap for public pages

## Current Tasks

- [x] get all public users and add them to the sitemap
- [x] register the sitemap to sitemap index

### CODE REVIEW

File: public/sitemaps.xml [hunk 1]
```diff
@@ -24,4 +24,8 @@
     <loc>https://www.tanyaaja.in/login</loc>
     <lastmod>2023-09-27</lastmod>
   </url>
+  <url>
+    <loc>https://www.tanyaaja.in/sitemap-public-users.xml</loc>
```
Comment (mazipan) [inline public/sitemaps.xml]: Oh lupa, belum ada sitemap index nya ternyata :-D

**Comment by mazipan** [comment]
Tambahin file sitemap-index.xml @adjieguntoro-tokped di dir `public/`

```xml
<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  <sitemap>
    <loc>https://www.tanyaaja.in/sitemaps.xml</loc>
    <lastmod>2023-10-20T17:00:00.000Z</lastmod>
  </sitemap>
  <sitemap>
    <loc>https://www.tanyaaja.in/sitemap-public-users.xml</loc>
    <lastmod>2023-10-20T17:00:00.000Z</lastmod>
  </sitemap>
</sitemapindex>
```

File: public/sitemaps.xml (hunk 1 above)
Comment (mazipan) [reply public/sitemaps.xml]: Ini harusnya di index sitemap nya nge-link nya.
Maaf2 pak, gak ngecek :-D

Comment (adjieguntoro-dev) [reply public/sitemaps.xml]: udah diremove ya..

**Review by mazipan — APPROVED** [review]
MANTAPPP, Seperti biasanya 🚀

**Comment by adjieguntoro-dev** [comment]
wah mas, ada yang kelewat, hoho... mapping nya lupa dijoin, masih ada comma nya..

**Pull request merged**